### PR TITLE
Persist worker_name after job is finished

### PR DESCRIPTION
**Pull request by eswolinsky3241**
Persisting the worker_name on the job object in Redis would allow for debugging and analyzing logs from the worker

### CODE REVIEW

**Comment by eswolinsky3241** [comment]
@selwin Thoughts on this change? We are running workers in kubernetes and would be great to be able to see the pod that exeuted the job after completion in case we need to check the logs or ssh into it.

File: rq/worker.py [hunk 1]
```diff
@@ -1060,8 +1059,8 @@ def handle_job_success(self, job: 'Job', queue: 'Queue', started_job_registry: S
                     if result_ttl != 0:
                         self.log.debug('Setting job %s status to finished', job.id)
                         job.set_status(JobStatus.FINISHED, pipeline=pipeline)
-                        job.worker_name = None
-
+                        # Don't clobber the user's meta dictionary!
+                        job.save(pipeline=pipeline, include_meta=False)
```
[the remark this answers is not in the document]
Comment (selwin) [reply rq/worker.py:1063]: There's already a `job.save()` a few lines below.

**Comment by selwin** [comment]
I'm ok with this change. Please see my comment, I think the previous commit was not merged cleanly.

**Comment by eswolinsky3241** [comment]
@selwin thanks, I removed that redundant line

**Comment by selwin** [comment]
@eswolinsky3241 mind fixing the failing tests before I merge this in?

**Comment by eswolinsky3241** [comment]
@selwin No problem, I removed the assertINull check for the worker_name attribute on the job.

**Comment by eswolinsky3241** [comment]
@selwin Looks like some new tests are failing, is this something I need to fix on my end?

**Comment by selwin** [comment]
The builds that failed are all based on Python 3.6. Try pingging the Ubuntu build to 20.04. More information here:  https://github.com/actions/setup-python/issues/544

**Comment by eswolinsky3241** [comment]
@selwin Thanks, I've pegged the runner to 20.04, mind running the workflows again?

**Comment by eswolinsky3241** [comment]
@selwin, just curious, any idea when the next release will be?

**Comment by selwin** [comment]
I'll probably make a release after the holidays. https://github.com/rq/rq/pull/1725 was merged two weeks ago, although it should be fully backwards compatible, I want to let it sit in master for a bit for people to try out before making it available for wider release.

So if you're waiting to use this in production soon, you should try pulling directly from a commit.

**Pull request merged**